src/appleseed/foundation/curve/binarycurvefilereader.cpp

```diff
@@ -0,0 +1,179 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Girish Ramesh, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "binarycurvefilereader.h"
+
+// appleseed.foundation headers.
+#include "foundation/core/exceptions/exceptionioerror.h"
+#include "foundation/math/vector.h"
+#include "foundation/curve/icurvebuilder.h"
+#include "foundation/image/color.h"
+#include "foundation/platform/types.h"
+#include "foundation/utility/bufferedfile.h"
+#include "foundation/utility/memory.h"
+
+// Standard headers.
+#include <cstring>
+#include <memory>
+
+using namespace std;
+
+namespace foundation
+{
+
+//
+// BinaryCurveFileReader class implementation.
+//
+
+BinaryCurveFileReader::BinaryCurveFileReader(const string& filename)
+  : m_filename(filename)
+{
+}
+
+void BinaryCurveFileReader::read(ICurveBuilder& builder)
+{
+    BufferedFile file(
+        m_filename.c_str(),
+        BufferedFile::BinaryType,
+        BufferedFile::ReadMode);
+
+    if (!file.is_open())
+        throw ExceptionIOError();
+
+    read_and_check_signature(file);
+
+    uint16 version;
+    checked_read(file, version);
+
+    unique_ptr<ReaderAdapter> reader;
+
+    switch (version)
```

Comment:
Ok, the way we indent `switch` statements is slightly peculiar:

The _body_ of `case` statements is indented once (4 spaces) with respect to the `switch` statement; `case` statements themselves are only _half-indented_. Here's an example:

    switch (version)
    {
      case 1:                   // half-indent (2 spaces)
        reader.reset(...);      // half-indent wrt. case, full indent wrt. switch
        break;
    }

The rationale behind this choice is to avoid over-indenting code.

It's the same thing as with `public`, `protected` and `private` statements: just like `case`, they don't warrant a full indentation.

Reply:
Fixed.